src/middleware.ts

```diff
@@ -12,6 +12,13 @@ export function middleware(request: NextRequest) {
 
   const requestHeaders = new Headers(request.headers);
   requestHeaders.set("x-nonce", nonce);
+
+  if (typeof process.env.E2E_TEST_ENV !== "undefined") {
```

Comment:
I'm worried that if this env var is set somehow in stage or prod, we would have this feature flag forced? 

Would it be better to have more guards like `&& not in prod && not in stage`?

Reply:
I was thinking about this as well and had the condition first only to check for `process.env.E2E_TEST_ENV === "local"`. Now you are mentioning it: Yes, let’s play it safe — changed in [4ae7859](https://github.com/mozilla/blurts-server/pull/5150/commits/4ae78594797e925b6e975b4b6ae4d73f019da948).